packages/hierarchy-browser/src/hooks/useCommunityDetails.ts

```diff
@@ -18,10 +19,19 @@ export function useCommunityLevelText(
 
 export function useCommunityLevelCalculator(
 	data: ICommunityDetail[],
-): [number, number] {
-	return useMemo(() => {
+): [number, number, ICommunity[]] {
+	const [min, max] = useMemo(() => {
 		return [0, data.length - 1]
 	}, [data])
+
+	const communityWithLevels = useMemo(() => {
+		const reverseList = [...data].reverse()
+		return reverseList.map(
+			(comm, index) =>
+				Object.assign({}, { ...comm, level: index }) as ICommunity,
```

Comment:
Replace with `{...comm, level: index}`. Object.assign and spread in tandem is like doing two object.assign statements